File: LEETCODE/LEETCODE-JAVA/704.java
//704. Binary Search
//https://leetcode.com/problems/binary-search/

class Solution {
    public int search(int[] nums, int target) {
        int n=nums.length;
        int s=0, e=n-1;
        
        while(s<=e){
            int mid=s+(e-s)/2;
            
            if(nums[mid]==target)
                return mid;
            else if(nums[mid]>target)
                e=mid-1;
            else
                s=mid+1;
        }
    
        return -1;
    }
}

File: LEETCODE/LEETCODE-PYTHON/704.py
# 704. Binary Search
# https://leetcode.com/problems/binary-search/

# Method-1 Binary Search
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low, high= 0, len(nums)-1

        while(low<=high):
            mid=low+(high-low)//2

            if nums[mid]==target:
                return mid
            elif nums[mid]<target:
                low=mid+1
            elif nums[mid]>target:
                high=mid-1

        return -1